retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp

```diff
@@ -1227,8 +1227,17 @@ void GxsForumThreadWidget::insertMessage()
 
     ui->versions_CB->blockSignals(false) ;
 
-    /* request Post */
-    updateMessageData(mThreadId);
+	/* request Post */
+	bool missing = index.sibling(index.row(),RsGxsForumModel::COLUMN_THREAD_DATA).data(RsGxsForumModel::MissingRole).toBool();
+	if (missing)
+	{
+		// Don't update data for missing message else get multiple entry.
+		setMessageLoadingError(tr("Missing Message:\nThis message is missing. You should receive it later."));

```

Comment:
You should => you may. Not sure about this. Maybe an english native speaker could advise here.

Reply:
You may receive => Vous pouvez recevoir
You should receive => Vous devriez recevoir
But maybe Google is wrong. ;)

Reply:
"You should" can  mean that there actually is something to do on your own to fix it. That what I find ambiguous about this sentence.